Fixed nil pointer issue during tx gas price estimation in keeper

begmaroman opened the pull request.

infiloop2 commented on core/services/keeper/upkeep_executer_test.go:
```diff
@@ -191,6 +191,58 @@ func Test_UpkeepExecuter_PerformsUpkeep_Happy(t *testing.T) {
+		config.Overrides.GlobalEvmEIP1559DynamicFees = null.BoolFrom(true)
+
+		gasLimit := upkeep.ExecuteGas + config.KeeperRegistryPerformGasOverhead()
+		//once you know where the root nil pointer dereference is youll need to fix these gas values
+		//so the mocks get what they expect based on GasPrice nil
```
Can you productionise the test please

begmaroman replied:
Tests are ready for review now.

infiloop2 commented on core/services/keeper/upkeep_executer_test.go:
```diff
@@ -191,6 +191,58 @@ func Test_UpkeepExecuter_PerformsUpkeep_Happy(t *testing.T) {
+	t.Run("runs upkeep on triggering block number on EIP1559 chain GlobalEvmEIP1559DynamicFees true", func(t *testing.T) {
+		db, config, ethMock, executer, registry, upkeep, job, jpv2, txm, _, _, _ := setup(t)
+
+		config.Overrides.GlobalEvmEIP1559DynamicFees = null.BoolFrom(true)
```
Do we have KeeperCheckUpkeepGasPriceFeatureEnabled = true by default?

begmaroman replied:
It is true by default. Just to be explicit.

infiloop2 reviewed (changes requested):
Can you also please verify that this test was failing before your change and now passes

begmaroman commented:
@infiloop2 verified, it panics. 

Actually, it should be tested by an already existing test but there is a bug - it does not enable EIP1559. Fixed an existing test and added a new one (both for EIP1559 and non-EIP1559).

infiloop2 commented on core/services/keeper/upkeep_executer_test.go:
```diff
@@ -144,9 +148,60 @@ func Test_UpkeepExecuter_PerformsUpkeep_Happy(t *testing.T) {
+	t.Run("runs upkeep on triggering block number on non-EIP1559 chain", func(t *testing.T) {
+		db, config, ethMock, executer, registry, upkeep, job, jpv2, txm, _, _, _ := setup(t)
+
+		config.Overrides.GlobalEvmEIP1559DynamicFees = null.BoolFrom(false)
```
the two tests look the same to me with the change of this config. Could you parameterise the two tests to share the code. (e.g. https://github.com/smartcontractkit/chainlink/blob/develop/core/cmd/remote_client_test.go#L151-L162)

begmaroman replied:
Well, it is kind of difficult to refactor all tests to the test-table approach. I've updated it to use the same function but with a different flag. Check again.

infiloop2 reviewed (approved):
looks good, thanks for fixing it properly

pinebit commented:
Should not this target 1.3.0-rc0 branch?
